### PR TITLE
Fix logging in the examples

### PR DESCRIPTION
**Pull request by jplu**
# What does this PR do?

This PR updates all the examples to use the Transformers logging util. Before no logs was displayed.

### CODE REVIEW

File: examples/language-modeling/run_clm.py [hunk 1]
```diff
@@ -43,6 +43,12 @@
     set_seed,
 )
 from transformers.trainer_utils import is_main_process
+from transformers.utils import logging as hf_logging
+
+
+hf_logging.enable_default_handler()
+hf_logging.set_verbosity_info()
+hf_logging.enable_explicit_format()
```
Comment (LysandreJik) [inline examples/language-modeling/run_clm.py:51]: Don't think we need that here as it's already handled below cc @sgugger

File: examples/language-modeling/run_mlm.py [hunk 2]
```diff
@@ -43,8 +43,13 @@
     set_seed,
 )
 from transformers.trainer_utils import is_main_process
+from transformers.utils import logging as hf_logging
 
 
+hf_logging.enable_default_handler()
+hf_logging.set_verbosity_info()
+hf_logging.enable_explicit_format()
+
```
Comment (LysandreJik) [inline examples/language-modeling/run_mlm.py:52]: Don't think we need that here as it's already handled below cc @sgugger

File: examples/language-modeling/run_mlm_wwm.py [hunk 3]
```diff
@@ -45,8 +45,13 @@
     set_seed,
 )
 from transformers.trainer_utils import is_main_process
+from transformers.utils import logging as hf_logging
 
 
+hf_logging.enable_default_handler()
+hf_logging.set_verbosity_info()
+hf_logging.enable_explicit_format()
+
```
Comment (LysandreJik) [inline examples/language-modeling/run_mlm_wwm.py:54]: Don't think we need that here as it's already handled below cc @sgugger

File: examples/language-modeling/run_plm.py [hunk 4]
```diff
@@ -39,8 +39,13 @@
     set_seed,
 )
 from transformers.trainer_utils import is_main_process
+from transformers.utils import logging as hf_logging
 
 
+hf_logging.enable_default_handler()
+hf_logging.set_verbosity_info()
+hf_logging.enable_explicit_format()
+
```
Comment (LysandreJik) [inline examples/language-modeling/run_plm.py:48]: Don't think we need that here as it's already handled below cc @sgugger

File: examples/test_examples.py [hunk 5]
```diff
@@ -24,8 +24,13 @@
 
 from transformers.file_utils import is_apex_available
 from transformers.testing_utils import TestCasePlus, require_torch_non_multi_gpu_but_fix_me, torch_device
+from transformers.utils import logging as hf_logging
 
 
+hf_logging.enable_default_handler()
+hf_logging.set_verbosity_info()
+hf_logging.enable_explicit_format()
+
```
Comment (LysandreJik) [inline examples/test_examples.py:33]: I don't think we need more logging in the tests

File: examples/test_xla_examples.py [hunk 6]
```diff
@@ -21,8 +21,13 @@
 from unittest.mock import patch
 
 from transformers.testing_utils import require_torch_non_multi_gpu_but_fix_me, require_torch_tpu
+from transformers.utils import logging as hf_logging
 
 
+hf_logging.enable_default_handler()
+hf_logging.set_verbosity_info()
+hf_logging.enable_explicit_format()
+
```
Comment (LysandreJik) [inline examples/test_xla_examples.py:30]: I don't think we need more logging in the tests

File: examples/text-classification/run_glue.py [hunk 7]
```diff
@@ -39,6 +39,12 @@
     set_seed,
 )
 from transformers.trainer_utils import is_main_process
+from transformers.utils import logging as hf_logging
+
+
+hf_logging.enable_default_handler()
+hf_logging.set_verbosity_info()
+hf_logging.enable_explicit_format()
```
Comment (LysandreJik) [inline examples/text-classification/run_glue.py:47]: I think this is already handled below cc @sgugger

**Review by LysandreJik — APPROVED** [review]
Great, LGTM! I think some of these shouldn't be here (see comments), but overall looks great! Thanks @jplu!

File: examples/language-modeling/run_clm.py (hunk 1 above)
Comment (jplu) [reply examples/language-modeling/run_clm.py:51]: Where is-it handled in the file? I don't see it.

File: examples/language-modeling/run_mlm.py (hunk 2 above)
Comment (jplu) [reply examples/language-modeling/run_mlm.py:52]: Where is-it handled in the file? I don't see it.

File: examples/language-modeling/run_mlm_wwm.py (hunk 3 above)
Comment (jplu) [reply examples/language-modeling/run_mlm_wwm.py:54]: Where is-it handled in the file? I don't see it.

File: examples/language-modeling/run_plm.py (hunk 4 above)
Comment (jplu) [reply examples/language-modeling/run_plm.py:48]: Where is-it handled in the file? I don't see it.

File: examples/text-classification/run_glue.py (hunk 7 above)
Comment (jplu) [reply examples/text-classification/run_glue.py:47]: Where is-it handled in the file? I don't see it.

File: examples/language-modeling/run_clm.py (hunk 1 above)
Comment (sgugger) [reply examples/language-modeling/run_clm.py:51]: Line 170

File: examples/language-modeling/run_mlm.py (hunk 2 above)
Comment (sgugger) [reply examples/language-modeling/run_mlm.py:52]: Line 181

File: examples/language-modeling/run_mlm_wwm.py (hunk 3 above)
Comment (sgugger) [reply examples/language-modeling/run_mlm_wwm.py:54]: Line 188

File: examples/language-modeling/run_plm.py (hunk 4 above)
Comment (sgugger) [reply examples/language-modeling/run_plm.py:48]: Line 178

File: examples/text-classification/run_glue.py (hunk 7 above)
Comment (sgugger) [reply examples/text-classification/run_glue.py:47]: Line 173

File: examples/language-modeling/run_clm.py (hunk 1 above)
Comment (jplu) [reply examples/language-modeling/run_clm.py:51]: Line 170 I have 
```
logger.warning(
        f"Process rank: {training_args.local_rank}, device: {training_args.device}, n_gpu: {training_args.n_gpu}"
        + f"distributed training: {bool(training_args.local_rank != -1)}, 16-bits training: {training_args.fp16}"
    )
```

Not related.

File: examples/language-modeling/run_mlm.py (hunk 2 above)
Comment (jplu) [reply examples/language-modeling/run_mlm.py:52]: Line 181 I have:

```
logger.warning(
        f"Process rank: {training_args.local_rank}, device: {training_args.device}, n_gpu: {training_args.n_gpu}"
        + f"distributed training: {bool(training_args.local_rank != -1)}, 16-bits training: {training_args.fp16}"
    )
```

Not related.

File: examples/language-modeling/run_mlm_wwm.py (hunk 3 above)
Comment (jplu) [reply examples/language-modeling/run_mlm_wwm.py:54]: Line 188 I have:
```
logger.warning(
        f"Process rank: {training_args.local_rank}, device: {training_args.device}, n_gpu: {training_args.n_gpu}"
        + f"distributed training: {bool(training_args.local_rank != -1)}, 16-bits training: {training_args.fp16}"
    )
```
Not related.

File: examples/language-modeling/run_plm.py (hunk 4 above)
Comment (jplu) [reply examples/language-modeling/run_plm.py:48]: Line 178 I have:
```
logger.warning(
        f"Process rank: {training_args.local_rank}, device: {training_args.device}, n_gpu: {training_args.n_gpu}"
        + f"distributed training: {bool(training_args.local_rank != -1)}, 16-bits training: {training_args.fp16}"
    )
```
Not related.

File: examples/text-classification/run_glue.py (hunk 7 above)
Comment (jplu) [reply examples/text-classification/run_glue.py:47]: Line 173 I have:
```
logger.warning(
        f"Process rank: {training_args.local_rank}, device: {training_args.device}, n_gpu: {training_args.n_gpu}"
        + f"distributed training: {bool(training_args.local_rank != -1)}, 16-bits training: {training_args.fp16}"
    )
```
Not related.

**Comment by LysandreJik** [comment]
The five examples scripts mentioned above all already use the huggingface logger, and already set it to the correct verbosity level. Seeing as the example scripts are *examples*, they really should be as straightforward as possible, and should not repeat unnecessary statements.

Please update the following files: `run_clm.py`, `run_mlm.py`, `run_glue.py`, `run_mlm_wwm.py`, `run_plm.py` so that they do not have useless statements.

@sgugger pointed you towards the correct area, you should be able to find it if you look for "logging".

**Comment by jplu** [comment]
I don't see the import neither the usage. Can you link it please?

**Comment by LysandreJik** [comment]
https://github.com/huggingface/transformers/blob/master/examples/language-modeling/run_clm.py#L170

**Comment by jplu** [comment]
Ok got it I was looking at my own updated file ahah.

**Comment by LysandreJik** [comment]
It is also in your file, below https://github.com/huggingface/transformers/blob/d410b83111238f7b949b7c9c6a4c3f689d29519b/examples/language-modeling/run_clm.py#L176

**Comment by jplu** [comment]
Ok I have removed the duplicate import and add only the two missing calls.

**Review by sgugger — CHANGES_REQUESTED** [review]
Most of the example scripts can run in a distributed fashion. The logging should not be set like this indiscriminately, but only on the main process (see the test `if is_main_process(training_args.local_rank):` in the new examples like `run_glue.py` for instance).

The current suggested changes will result in every process printing the logs which is not a behavior we want either.

**Comment by jplu** [comment]
Already done the changes 👍

**Comment by sgugger** [comment]
No, the changes are not done:
1. in the scripts like the new `run_glue.py`, the code for logging is duplicated and executed once for every process, then once for the main process.
2. in all the other scripts, the code is executed on all processes.

This is also added in util files or test files where it's just not necessary.

**Comment by jplu** [comment]
I really don't get what you mean, sorry. What are the exact changes I have to make for each file?

**Comment by jplu** [comment]
@sgugger Please check the last commit and let me know if it was what you meant. Otherwise, please, can you be more specific in what you want me to change.

**Review by LysandreJik — APPROVED** [review]
I think this looks good. Thank you for iterating and doing the changes.

Offline approval from @sgugger, so merging.

**Pull request merged**